src/internal/interpreter/http/http.ts

```diff
@@ -170,101 +234,91 @@ export class HttpClient {
       }
 
       if (configuration.type === SecurityType.APIKEY) {
-        applyApiKeyAuth(contextForSecurity, configuration);
+        const handler = new ApiKeyHandler()
+        handler.prepare(contextForSecurity, configuration)
```

Comment:
if you don't need to do stuff with the constructed unprepared handler, it'll be probably better to use static prepare for construction, i.e. `const handler = ApiKeyHandler.prepare(contextForSecurity, configuration)`

Reply:
You mean the `ApiKeyHandler.prepare(contextForSecurity, configuration)` should change the `contextForSecurity` and return new `ApiKeyHandler` instance? It feels like two different things. But maybe the "change the `contextForSecurity`" part is hard to spot at first 🤔

Reply:
oh i didn't notice it changes the context. that's not not very good tbh :thinking:

Reply:
but it's okay now, with the configuration in constructor :shrug: